### DISCUSSION TITLE
Help: Vercel deployment failing with "document is not defined" error for Deepgram Voice Agent Demo

### DISCUSSION THREAD

**Discussion by btconway** [deepgram/community#1029] [General help]
Hi! I'm trying to deploy the Deepgram Voice Agent Demo (https://github.com/deepgram-devs/deepgram-voice-agent-demo) on Vercel but running into build errors during deployment.

The build fails with:

ReferenceError: document is not defined
at createTag (/vercel/path0/.next/server/chunks/567.js:1:8756)

Details:
- Using Next.js version 14.2.18
- Following the README setup instructions
- Created the necessary .env file with DEEPGRAM_API_KEY
- Local development with `yarn dev` works fine
- The error occurs during Vercel's static page generation for "/" and "/_not-found" routes
- Build compiles successfully but fails during static export

I'm not a frontend developer, so I'd appreciate help understanding what might be causing this error. From what I've read, it seems related to trying to access the browser's `document` object during server-side rendering - is there a configuration I need to adjust for Vercel deployment?

Full build log attached below for reference.

[vercel_build_log.txt](https://github.com/user-attachments/files/18181566/vercel_build_log.txt)

**Comment by deepgram-community** [comment]
Thanks for asking your question. Please be sure to reply with as much detail as possible so the community can assist you efficiently.
<sub>Consider joining our [Discord community](https://discord.gg/deepgram) for more opportunity to engage with your fellow Deepgram users. You can earn points which can be redeemed for cool stuff by being active in our communities!</sub>

**Comment by deepgram-community** [comment]
Hey there! It looks like you haven't connected your GitHub account to your Deepgram account. You can do this at https://community.deepgram.com - being verified through this process will allow our team to help you in a much more streamlined fashion.

**Comment by deepgram-community** [comment]
It looks like we're missing some important information to help debug your issue. Would you mind providing us with the following details in a reply?

- The deepgram product you are using (e.g Speech to Text, Agent API)
- A request ID that triggered your error or issue.

**Reply by btconway** [reply]
- Deepgram Voice Agent
